src/lxc/monitor.c

```diff
@@ -181,7 +181,6 @@ int lxc_monitor_sock_name(const char *lxcpath, struct sockaddr_un *addr) {
 	if (ret < 0)
 		return -1;
 
-	sockname[sizeof(addr->sun_path)-3] = '\0';
```

Comment:
-2 for the first and last \0 byte and to actually write a \0 byte you need to do -3.